### ISSUE TITLE
Tracking: GCC 8.2

### ISSUE THREAD

**Issue by chneukirchen** [void-linux/void-packages#1525]
This issue resumes https://github.com/voidlinux/void-packages/issues/14633.

gcc 8.2 is out now, and we should see which remaining packages are broken and how to fix them.

**Comment by pullmoll** [comment]
I will move my current set of packages built to a safe place and thenstart another cycle of bootstrapping with gcc-8.2 to see how far I get.

**Comment by chneukirchen** [comment]
Arch uses isl 0.19.

**Comment by pullmoll** [comment]
Ok, I'll got with `isl-0.19` then and update `isl15` for `cross-*` as well.

**Comment by pullmoll** [comment]
Packages which fail to build OOTB:
- [ ] grub-2.02
  In file included from ../../grub-core/efiemu/runtime/efiemu.c:33:
../../include/grub/efiemu/runtime.h:36:1: error: alignment 1 of 'struct efi_variable' is less than 8 [-Werror=packed-not-aligned]
 } GRUB_PACKED;
 ^
- [ ] ...

**Comment by chneukirchen** [comment]
Is the bumped branch public? Then I can debug locally.

**Comment by chneukirchen** [comment]
Fix at https://git.archlinux.org/svntogit/packages.git/tree/trunk/0008-Fix-packed-not-aligned-error-on-GCC-8.patch?h=packages/grub&id=0c7d22e66884c85574eca8ce7df9571748aabb29

**Comment by pullmoll** [comment]
The bumped branch is in [my fork's master](https://github.com/pullmoll/void-packages/).
The patch works fine and I'll add a commit.

**Comment by pullmoll** [comment]
base-chroot{,-musl} for the 4 native architectures is completed. Now building base-system.

**Comment by chneukirchen** [comment]
librsvg-devel-2.40.20_4 doesn't contain vala stuff here oO.

I guess this should be moved to a separate package then.

**Comment by chneukirchen** [comment]
The vala stuff resides in librsvg, where it doesn't belong.

**Comment by pullmoll** [comment]
cross-arm-none-eabi needs a fix for the patch to enable. I tried it, but it fails because of some part of the build requiring hard float for arm? I guess that needs to be fixed by its maintainer @teajay-fr .

**Comment by pullmoll** [comment]
architecture | packages built
----------------|--------------------
aarch64-musl | 274
aarch64 | 284
armv5tel-musl | 237
armv5tel | 286
armv6l-musl | 288
armv6l | 286
armv7l-musl | 288
armv7l | 286
i686-musl | 498
i686 | 1029
mips-musl | 231
mipsel-musl | 28
mipselhf-musl | 28
mipshf-musl | 28
x86_64-musl | 1012
x86_64 | 2338

**Comment by pullmoll** [comment]
@chneukirchen I don't understand. `srcpkgs/librsvg/template` has `if [ "$build_option_vala" ]; then` which should be right, no? Yet there is no `usr/share/vala` data in `librsvg-devel`... strikes me a bit odd.

**Comment by chneukirchen** [comment]
No, the variable is spelt wrong with an `s`.

**Comment by pullmoll** [comment]
New issue with linux4.14 and plugins:
```
Cannot use CONFIG_GCC_PLUGINS: your gcc installation does not support plugins, perhaps the necessary headers are missing?
make: *** [scripts/Makefile.gcc-plugins:70: gcc-plugins-check] Error 1
=> ERROR: linux4.14-4.14.61_1: do_build: 'make ARCH=$arch ${_version} ${_cross} ${makejobs} prepare' exited with 2
=> ERROR:   in do_build() at srcpkgs/linux4.14/template:95
```
IMHO we really should get rid of the plugin stuff for the kernels.

**Comment by chneukirchen** [comment]
Yes, let's do it.

**Comment by pullmoll** [comment]
@teajay-fr : no problem. The package is marked as broken for the update, so it will be built only after you found time to do whatever is necessary to build it for gcc-8.2.0.

**Comment by thypon** [comment]
Can we fallback to disable GCC plugins only in case of errors?

**Comment by pullmoll** [comment]
@thypon I guess you meant to add `--enable-plugins` to gcc's template? I think that's ok. Opinions?

**Comment by pullmoll** [comment]
TODOs:
- [ ] revbump linux3.16
- [ ] revbump or remove linux3.18 (EOL)
- [ ] revbump or remove linux4.1 (EOL)
- [ ] drop linux4.15
- [ ] drop linux4.16
- [x] revbump linux4.14
- [x] revbump linux4.17
- [ ] revbump linux4.18

- [ ] hope that the musl cross builders pickup libgfortran.so.5 at some point and continue to build blas etc.

**Comment by mvf** [comment]
#1799 fixes the `mozc` build with GCC 8.2

**Comment by pullmoll** [comment]
cross-vpkg-dummy also needs a host build on the aarch64 builders
